components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/NewApiVersionMailNotifier.java

```diff
@@ -0,0 +1,150 @@
+            APIManagementException {
+
+        Properties props = new Properties();
+        props.put("mail.smtp.host", SMTP_HOST_NAME);
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.smtp.starttls.enable", "true");
+        props.put("mail.smtp.port", "587");
+        props.put("mail.transport.protocol", "smtp");
+        //get Notifier email List
+        Set<String> sendTo = getEmailNotifierList(notificationDTO);
+
+        if (sendTo.isEmpty()) {
+            log.debug("Email Notifier Set is Empty");
+            return;
+        }
+        for (String mail : sendTo) {
+            try {
+                Authenticator auth = new SMTPAuthenticator();
+                Session mailSession = Session.getDefaultInstance(props, auth);
+                MimeMessage message = new MimeMessage(mailSession);
+                notificationDTO.setTitle((String) notificationDTO.getProperty(NotifierConstants.TITLE_KEY));
+                notificationDTO.setMessage((String) notificationDTO.getProperty(NotifierConstants.TEMPLATE_KEY));
+                notificationDTO = loadMailTemplate(notificationDTO);
+                message.setSubject(notificationDTO.getTitle());
+                message.setContent(notificationDTO.getMessage(), NotifierConstants.TEXT_TYPE);
+                message.setFrom(new InternetAddress(mailConfigurations.getFromUser()));
+
+                message.addRecipient(Message.RecipientType.TO, new InternetAddress(mail));
+
+                Transport.send(message);
+
+            } catch (MessagingException e) {
+                log.error("Exception Occurred during Email notification Sending", e);
+            }
+
+        }
+    }
+
+    /**
+     * Get UserNotifier Email Set.
+     */
+    private Set<String> getEmailNotifierList(NotificationDTO notificationDTO) throws
+            APIManagementException {
+        Set<String> mailSet = new HashSet<>();
+        Set<String> subscriberList = (Set<String>) notificationDTO.
+                getProperty(NotifierConstants.SUBSCRIBERS_PER_API);
+        if (!subscriberList.isEmpty()) {
+            for (Iterator<String> it = subscriberList.iterator(); it.hasNext(); ) {
+                String userName = it.next();
+                APIManagerFactory apiManagerFactory = APIManagerFactory.getInstance();
+                String userId = apiManagerFactory.getIdentityProvider().getIdOfUser(userName);
+                mailSet.add(apiManagerFactory.getIdentityProvider().getEmailOfUser(userId));
+            }
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Subscriber List is Empty");
+            }
+        }
+        return mailSet;
+    }
+
+    /**
+     * Class to Authenticate User.
+     */
+    private static class SMTPAuthenticator extends Authenticator {
+        public PasswordAuthentication getPasswordAuthentication() {
+            String username = SMTP_AUTH_USER;
+            String password = SMTP_AUTH_PWD;
```

Comment:
for passwords we have to maintain a charArray

Reply:
PasswordAuthentication(username, password) method doesn't support for char array. so if that change to char[] then need to convert back string inside that.is it ok??